test/datapipes/test_mesh_datapipe.py

```diff
@@ -0,0 +1,172 @@
+
+
+@pytest.fixture
+def cgns_data_dir():
+    path = "/data/nfs/modulus-data/datasets/sample_formats"
+    return path
+
+
+def _create_random_vtp_vtu_mesh(num_points: int, num_triangles: int, dir: str) -> tuple:
+    """
+    Create a random VTP (VTK PolyData) mesh and a random VTU (VTK Unstructured Grid) mesh with triangles.
+
+    Parameters:
+        num_points (int): Number of random points.
+        num_triangles (int): Number of triangles.
+        dir (str): Directory to save the VTP and VTU files.
+
+    Returns:
+        tuple: A tuple containing the random VTP mesh (vtk.vtkPolyData) and the random VTU mesh (vtk.vtkUnstructuredGrid).
+    """
+
+    # make directory if it does not exist
+    os.makedirs(dir, exist_ok=True)
+
+    # Create random points
+    points = vtk.vtkPoints()
+    for _ in range(num_points):
+        x, y, z = (
+            random.uniform(-10, 10),
+            random.uniform(-10, 10),
+            random.uniform(-10, 10),
+        )
+        points.InsertNextPoint(x, y, z)
+
+    # Create triangles
+    triangles = vtk.vtkCellArray()
+    for _ in range(num_triangles):
+        p1, p2, p3 = (
+            random.randint(0, num_points - 1),
+            random.randint(0, num_points - 1),
+            random.randint(0, num_points - 1),
+        )
+        triangle = vtk.vtkTriangle()
+        triangle.GetPointIds().SetId(0, p1)
+        triangle.GetPointIds().SetId(1, p2)
+        triangle.GetPointIds().SetId(2, p3)
+        triangles.InsertNextCell(triangle)
+
+    # Create a PolyData object (VTP mesh)
+    vtp_mesh = vtk.vtkPolyData()
+    vtp_mesh.SetPoints(points)
+    vtp_mesh.SetPolys(triangles)
+
+    # Assign random scalar values (features) to points in VTP mesh
+    scalar_values = vtk.vtkDoubleArray()
+    scalar_values.SetName("RandomFeatures")
+    for _ in range(num_points):
+        scalar_values.InsertNextValue(random.uniform(0, 1))
+    vtp_mesh.GetPointData().SetScalars(scalar_values)
+
+    # Write VTP mesh to file
+    vtp_writer = vtk.vtkXMLPolyDataWriter()
+    vtp_writer.SetFileName(os.path.join(dir, "random.vtp"))
+    vtp_writer.SetInputData(vtp_mesh)
+    vtp_writer.Write()
+
+    # Create an Unstructured Grid (VTU mesh)
+    vtu_mesh = vtk.vtkUnstructuredGrid()
+    vtu_mesh.SetPoints(points)
+    vtu_mesh.SetCells(vtk.VTK_TRIANGLE, triangles)
+
+    # Assign random scalar values (features) to points in VTU mesh
+    scalar_values = vtk.vtkDoubleArray()
+    scalar_values.SetName("RandomFeatures")  # Set the name
+    for _ in range(num_points):
+        scalar_values.InsertNextValue(random.uniform(0, 1))
+    vtu_mesh.GetPointData().SetScalars(scalar_values)
+
+    # Write VTU mesh to file
+    vtu_writer = vtk.vtkXMLUnstructuredGridWriter()
+    vtu_writer.SetFileName(os.path.join(dir, "random.vtu"))
+    vtu_writer.SetInputData(vtu_mesh)
+    vtu_writer.Write()
+
+
+@pytest.mark.parametrize("device", ["cuda", "cpu"])
+def test_mesh_datapipe(device, tmp_path):
```

Comment:
Would be great if we can add the package check decorator for these similar to here: https://github.com/NVIDIA/modulus/blob/main/test/datapipes/test_ahmed_body.py#L33 to ensure smoother experience for the pip installers

Reply:
Done